stdlib/Artifacts/src/Artifacts.jl

```diff
@@ -496,8 +496,10 @@ function _artifact_str(__module__, artifacts_toml, name, path_tail, artifact_dic
 
     # If not, we need to download it.  We look up the Pkg module through `Base.loaded_modules()`
     # then invoke `ensure_artifact_installed()`:
-    Pkg = first(filter(p-> p[1].name == "Pkg", Base.loaded_modules))[2]
-    return jointail(Pkg.Artifacts.ensure_artifact_installed(string(name), artifacts_toml; platform), path_tail)
+    if ensure_artifact_installed === nothing
+        error("Artifact $(repr(name)) was not installed correctly. Try `using Pkg; Pkg.build(\"$__module__\"); Pkg.Artifacts.ensure_all_artifacts_installed($(repr(artifacts_toml)))`?")
```

Comment:
Shouldn't this error tell the package author to add `using Pkg` to their package?

Reply:
That would be telling them to prefer worse behavior. However, I like staticfloat's suggestion of examining the toml to figure out which was probably intended.